### ISSUE TITLE
Feature Request - Command Line Switch to Auto-liberate new content

### ISSUE THREAD

**Issue by kspillane** [rmcrackan/Libation#106]
Hey there!

I really like the program, much more usable than the others. Not sure if you are taking feature requests or even if this is something you would see as a viable feature, but I would love to be able to use a command-line launch argument to not open the gui and just scan all libraries and liberate anything new.

I get at least two or three books per month from audible, and while I could easily just open the program, make a couple clicks, and get the new books, the lazy admin in me would love to just setup a scheduled task to do it automatically.

**Comment by wtanksleyjr** [comment]
Ooh, I'd use that.

OTOH, I also wouldn't complain if I could do the same thing with a few
button clicks -- right now it's "click import, wait 30+ minutes; click
Export, switch to json, type 'books.json', click 'yes I want to overwrite
it', click Liberate, leave the room." It would be really cool if all of
that were "Click Liberate, check the import box, check the export box,
click OK, leave the room."

-wm

**Comment by rmcrackan** [comment]
I put in some prelim work on this today. I won't commit to a timeframe. It seems doable though. I'm going to leave this issue open while I think on this.

**Comment by kspillane** [comment]
Appreciate it! Figured it shouldn't be too bad, just triggering event
handlers in a different way. I'd help code, but never really got to
learning C#. I just stick to automation scripts in bash and python. :/

Let me know if you need any testing done.

Thanks again!

**Comment by rmcrackan** [comment]
Libationcli.exe is in latest release :) I'll update the readme soon-ish. Here's the summary:

Warnings about relying solely on on the CLI:
* CLI will not perform any upgrades.
* It will show that there is an upgrade, but that will likely scroll by too fast to notice.
* It will not perform all post-upgrade migrations. Some migrations are only be possible by launching GUI.

```
help
  libationcli --help
verb-specific help
  libationcli scan --help
scan all libraries
  scan
scan only libraries for specific accounts
  scan nickname1 nickname2
convert all m4b files to mp3
  convert
liberate all books and pdfs
  liberate
liberate pdfs only
  liberate --pdf
  liberate -p
export library to file
  export --path "C:\foo\bar\my.json" --json
  export -p "C:\foo\bar\my.json" -j
```

Currently logs are written to Console and to file. This means they'll be printed in the CLI. To disable, find this in Settings.json and delete the 3 lines after `"WriteTo": [`

```
  "Serilog": {
    "MinimumLevel": "Information",
    "WriteTo": [
      {
        "Name": "Console"
      },
```

**Comment by rmcrackan** [comment]
@kspillane @wtanksleyjr  Heads up: my current way of deploying the CLI is causing hell with the main app but only for some users. Soon I'm going to remove it from the release packages until I can think of a better way to deploy. This will not affect the current version of the CLI which works for you.

**Comment by wtanksleyjr** [comment]
Thanks for the heads up! I'll make a backup so I can keep that. It was an
amazingly cool hack how quickly you were able to add it, but it's 100% not
worth getting in the way of the main functionality.

-Wm

**Comment by kspillane** [comment]
No worries!
I meant to reply back sooner, I was curious, was there a way to liberate only new books? Or do the CLI and main app use different dbs? Or did I miss something? Otherwise each time I run it with the liberate option, it will re-liberate everything in my library.

Gonna pick up some new stuff tomorrow during the sale, let me know if you need any testing.

Thanks again!

**Comment by rmcrackan** [comment]
Same db. Same settings. :)

**Comment by rmcrackan** [comment]
> Thanks for the heads up! I'll make a backup so I can keep that. It was an
> amazingly cool hack how quickly you were able to add it, but it's 100% not
> worth getting in the way of the main functionality.

Thank you for your understanding. I admit being pretty proud of pounding that out so fast. It's depressing that this new setup acts so weird.

Regarding backups: I think Github keeps releases indefinitely. The CLI is available in all [releases](https://github.com/rmcrackan/Libation/releases/) from [v5.6.8](https://github.com/rmcrackan/Libation/releases/tag/v5.6.8) to [v5.7.3](https://github.com/rmcrackan/Libation/releases/tag/v5.7.3).

**Comment by rmcrackan** [comment]
@kspillane @wtanksleyjr : In [the latest release: v5.7.4](https://github.com/rmcrackan/Libation/releases/tag/v5.7.4) I figured out a new way to repackage which still includes Libation.exe and LibationCLI.exe in the same folder so they can share settings, db, etc. Per the release notes, since this issue was intermittent across users, I can't be 100% sure it's fixed. Please let me know if either program does not run for you.

**Comment by wtanksleyjr** [comment]
OK, apologies for the delay, I was able to test the new release both in GUI
and CLI, and it's EXCELLENT for my purposes.

I've got my whole workflow as a batch file now (sync-export-liberate), and
the GUI's available for when I want to single out a book or something. I
just used it, for example, to "ignore" all of the wall-street-journals
after I'd used the CLI to make absolutely sure there weren't any new books
that might get ignored if I did an import and something went wrong. WOOT!

-Wm
